src/db/spanner/batch.rs

```diff
@@ -433,20 +434,48 @@ pub async fn do_append_async(
             .clone()
             .incr("storage.spanner.adding_updates_to_batch_bsos");
         for val in update {
-            db.sql(
-                "UPDATE batch_bsos SET sortindex=@sortindex, payload=@payload, ttl=@ttl
+            let mut fields = Vec::new();
+            let mut sort_index = val.sortindex.get_string_value().to_owned();
+            if !sort_index.is_empty() {
+                fields.push("sortindex");
+            } else {
+                // This is not written, but the sql builder requires a numeric string
```

Comment:
What requires this value here (I don't understand about the sql builder requiring it). I suspect it's params and maybe param_types? We could build those dynamically (and or pop from param_types since it's already set up front)

Reply:
As far as I can tell, before the SQL is actually processed, the query builder resolves the types of the arguments. The problem is that the NoneType resolves to "" which fails to resolve to a numeric. That's why I pass in the "0", which does.

If I was SUPER clever, I would probably just exclude the values from the params using the same crafting technique I'm doing to build the sql, but that feels like making things really complicated rather than just do this, particluarly with the way that params have to be added.